packages/protocol/contracts/L1/libs/LibEthDepositing.sol

```diff
@@ -38,7 +38,8 @@ library LibEthDepositing {
 
         TaikoData.EthDeposit memory deposit = TaikoData.EthDeposit({
             recipient: msg.sender,
-            amount: uint96(msg.value)
+            amount: uint96(msg.value),
+            id: uint64(state.ethDeposits.length)
```

Comment:
`uint64(state.ethDeposits.length + 1)` is 0 allow here by the client?

Reply:
`id == 0` is allowed, since right now all withdrawal's id is `0`. but i believe the potential issue of this approach is we can not do that optimization any more? https://github.com/taikoxyz/taiko-mono/issues/13783 , which seems is quite necessary?

Reply:
We can still do it, but when it's done, we need some changes, but it's OK.